follow & favorite buttons

DorisW3 opened the pull request.

DorisW3 committed:
add favorite button component an dsiplay it on overview & my art page

DorisW3 committed:
bugfix

DorisW3 committed:
toggle favorite button

DorisW3 committed:
add follow button on overview page, add basic styling for button

DorisW3 committed:
add follow button on overview page, add basic styling for button

DorisW3 committed:
add user logo before username on overview page

MerleMak commented on pages/_app.js:
```diff
@@ -6,13 +6,17 @@ import useLocalStorageState from "use-local-storage-state";
   function handleAddEntry(newEntry) {
     setEntriesList([newEntry, ...entriesList]);
   }
+  // -----
+
+  // ----- favorite button
```
```suggestion
```

MerleMak commented on components/FavoriteButton.js:
```diff
@@ -0,0 +1,32 @@
+import { styled } from "styled-components";
+import { useState } from "react";
+
+export default function FavoriteButton({}) {
```
```suggestion
export default function FavoriteButton({isFavourite, handleToggle}) {
```

MerleMak replied:
Du musst die Props hier rausreichen, wenn du sie in der PictureList weiterverarbeitest :)

MerleMak commented on components/Overview.js:
```diff
@@ -3,12 +3,22 @@ import { styled } from "styled-components";
+import FollowButton from "./FollowButton.js";
 
 export default function Overview() {
+  // const [filledHeart, setFilledHeart] = useState(false);
+
+  /* function handleToggle() {
+    setFilledHeart(!filledHeart);
+  } */
+
```
```suggestion
```

MerleMak commented on components/PictureDetails.js:
```diff
@@ -4,14 +4,16 @@ import { styled } from "styled-components";
-  const routerId = router.query.index; //  gibt IMMER eine String zurück!!!!
+  const routerId = router.query.index; //  gibt IMMER eine String zurück!!!! id auch besser als String, da kein mathematischer Anwendungsfalls vorhanden
 
   const selectedPicture = pictures.find((picture) => picture.id === routerId);
 
   if (!selectedPicture) {
     return <p>Loading...</p>;
   }
+  // -----
```
```suggestion
  const selectedPicture = pictures.find((picture) => picture.id === routerId);

  if (!selectedPicture) {
    return <p>Loading...</p>;
  }
```

DorisW3 replied:
Hey Merle, hier verstehe ich dein Kommentar leider nicht. Was ist den damit gemeint?

MerleMak replied:
oh einfach nur den aus kommentierten code löschen also das `// ----`
Nichts wichtiges :)

MerleMak commented on components/PictureList.js:
```diff
@@ -3,14 +3,19 @@ import { styled } from "styled-components";
+import FavoriteButton from "./FavoriteButton.js";
 
-export default function PictureList() {
+export default function PictureList({ handleToggle, isFavorite }) {
```
Ich kann die index.js nicht sehen. Verwendest du diese Probs irgendwo weiter?

DorisW3 replied:
ich gebe sie noch an die index.js der myart page weiter, wo die PictureList aufgerufen wird

MerleMak commented on components/FollowButton.js:
```diff
@@ -0,0 +1,25 @@
+import { styled } from "styled-components";
+
+export default function FollowButton() {
+  const [follow, setFollow] = useState(false);
```
```suggestion
  const [isFollowing, setIsFollowing] = useState(false);
```

MerleMak commented on components/FollowButton.js:
```diff
@@ -0,0 +1,25 @@
+  return (
+    <StyledButton
+      onClick={() => {
+        setFollow(!follow);
```
```suggestion
        setIsFollowing(!isFollowing);
```

MerleMak commented on components/FollowButton.js:
```diff
@@ -0,0 +1,25 @@
+        setFollow(!follow);
+      }}
+    >
+      {follow ? "unfollow" : "follow"}
```
```suggestion
      {isFollowing ? "unfollow" : "follow"}
```

MerleMak reviewed (commented):
Es sieht gut aus! Aber ich sehe noch nicht, wo die die Information "isFavorite" und "isFollowing" global gespeichert wird?
Ich glaube es ist Sinnvoll wenn du auf Höhe der index.js oder der _app.js einen State Anlegst, wo diese User informationen gespeichert werden :)

DorisW3 committed:
Update pages/_app.js

Co-authored-by: Merle Makoben <merlemakoben@gmail.com>

DorisW3 committed:
Update components/FavoriteButton.js

wenn ich das mache, dann gibt es ein Fehler, weil ich beides in der Funktion deklariert wird

Co-authored-by: Merle Makoben <merlemakoben@gmail.com>

DorisW3 committed:
Update components/Overview.js

Co-authored-by: Merle Makoben <merlemakoben@gmail.com>

DorisW3 committed:
Update components/FollowButton.js

Co-authored-by: Merle Makoben <merlemakoben@gmail.com>

DorisW3 committed:
Update components/FollowButton.js

Co-authored-by: Merle Makoben <merlemakoben@gmail.com>

DorisW3 committed:
Update components/FollowButton.js

Co-authored-by: Merle Makoben <merlemakoben@gmail.com>

DorisW3 committed:
update pullrequest

DorisW3 committed:
Merge branch 'feat/fav-follow-buttons' of github.com:DorisW3/capstone-project into feat/fav-follow-buttons

DorisW3 committed:
fix merge-conflict

MerleMak reviewed (commented):
Okay, also ich denke du solltest das endlich mergen dürfen! :D
Aber ich habe einen Request für die nächste Userstory die du bearbeitest:
Bitte schreibe eine Userstory für das Speichern der isFavorite und isFollowing Daten im Local Storage :) 
Dieses Feature ist nötig, damit die Buttons einen echten Nutzen für das Favorite und Follow Feature haben. 
Dafür musst du
- die notwendigen Props auf der jeweiligen Button componente rausgeben
- diese props bis zur richtigen Page weitergeben
- die Information aus dem Local Storage in alle weiteren Componeten runterreichen, für die die Infos "isFollowing" und "isFavorite" relevant sind. (z.B. PictureDetail.js)
